### ISSUE TITLE
[MPPI] xtensor::optimize not found

### ISSUE THREAD

**Issue by kenloumixx** [ros-planning/navigation2#3537]
## Bug report

**Required Info:**

- Operating System: ubuntu 18.04
  - 
- ROS2 Version:  humble
  - 
- Version or commit hash: from source
  - 
- DDS implementation: 
  - 

#### Steps to reproduce issue

```
colcon build --symlink-install 
```

#### Expected behavior
Installed
#### Actual behavior
```
[Processing: nav2_mppi_controller]                                   
--- stderr: nav2_mppi_controller                                     
CMake Error at CMakeLists.txt:78 (target_link_libraries):
  Target "mppi_controller" links to:

    xtensor::optimize

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at CMakeLists.txt:78 (target_link_libraries):
  Target "mppi_critics" links to:

    xtensor::optimize

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

```
#### Additional information
I install the xtensor to use the [navigation2 package](https://github.com/ros-planning/navigation2/blob/main/nav2_mppi_controller/CMakeLists.txt#L78:~:text=target_link_libraries(%24%7Blib%7D%20xtensor%20xtensor%3A%3Aoptimize%20xtensor%3A%3Ause_xsimd)) in ROS. 
In the building process, I encounter the above message.
I installed the xtl and xtensor following [the guide](https://github.com/xtensor-stack/xtensor#install-from-sources). However, I don't know why this error occurs. 
Thank you in advance! 

----

**Comment by kenloumixx** [comment]
Close this issue as this is solved by right installation

**Comment by baronlq** [comment]
As I know, [xsimd] is defined as " not available" for my OS version (ubuntu 20.04). Hence, I could not build the "nav2_mppi_controller". I think 22.04 can solve this.

**Comment by SteveMacenski** [comment]
Correct, you can install it yourself on 20.04 pretty easily (I did that myself at one point while developing) but there aren't binaries available for some of those packages in older linux OS distributions
